embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/SupportedEnvelopeType.kt

```diff
@@ -0,0 +1,28 @@
+package io.embrace.android.embracesdk.internal.delivery
+
+import io.embrace.android.embracesdk.internal.payload.Envelope
+import java.lang.reflect.Type
+
+/**
+ * Enumerates the different types of telemetry that are supported when persisting to disk.
+ */
+enum class SupportedEnvelopeType(
```

Comment:
We only have two types of payloads that we wrap in our envelope - `SessionPayload` and `LogPayload`. The latter can contain a different mixture of logs, but they have the same shape. What is different is how each is prioritized and whether they are batched (and batching isn't really a concern of the delivery layer). 

So instead of having different types based on content, perhaps we should divide these along the lines of the actual payload type, and the priority of delivery? Right now, we have Crash, Log, and Network all being essentially LogPayloads with different priorities - but if we have another one later, say InternalErrors, I don't think we need another enum value if we just had various priorities, e.g. 

```
enum class DeliveryPriority(
  payloadType: Class
) {
  HIGHEST(LogPayload::class.java),
  HIGH(SessionPayload::class.java),
  NORMAL(LogPayload::class.java),
  LOW(LogPayload::class.java),
}
```

I wouldn't mind separating the payload type and the priority concepts all together as well, but not mapping one to one to the "types of telemetry in the thing" is what I'm going for.

Reply:
Is there a difference in that approach from what already exists other than semantics? I'll merge for now & we can discuss it in further detail

Reply:
Yeah I'll ping you. The idea is that if we have 10 variations of payloads with the same two structures, we won't have have 10 different constants to know how to sort between.